### PR TITLE
Spec: Add GCS and ADLS configuration to REST table load

### PR DESCRIPTION
**Pull request by Buktoria**
Adding missing documentation to the open-api spec for LoadTableResult configs for FileIO implementations GCS and ADLS.

### CODE REVIEW

File: open-api/rest-catalog-open-api.yaml [hunk 1]
```diff
@@ -2760,6 +2760,31 @@ components:
...
+         - `gcs.channel.read.chunk-size-bytes`: Optional INT that sets the minimum size that will be read by a single RPC. Read data will be locally buffered until consumed
+         - `gcs.channel.write.chunk-size-byte`: Optional INT that sets the  minimum size that will be written by a single RPC. Written data will be buffered and only flushed upon reaching this size or closing the channel.
+         - `gcs.oauth2.token`: String representation of the access token used for temporary access.
+         - `gcs.oauth2.token-expires-at`: A LONG that represents the date in Epoch milliseconds of when the token expires.
+         - `gcs.no-auth`: Boolean to explicitly configure "no authentication" for testing purposes using a GCS emulator
+         - `gcs.delete.batch-size`: Optional INT that configures the batch size used when deleting multiple files from a given GCS bucket. Defaults to 50.
+        
+        ## Azure Configurations
+        
+        The following configuration respected by the GCSFileIO
```
Comment (mrcnc) [inline open-api/rest-catalog-open-api.yaml:2781]: I think you have a copy/paste error here
```suggestion
        The following configuration respected by the ADLSFileIO
```

Comment (Buktoria) [reply open-api/rest-catalog-open-api.yaml]: whoops

File: open-api/rest-catalog-open-api.py [hunk 2]
```diff
@@ -1100,6 +1100,31 @@ class LoadTableResult(BaseModel):
      - `s3.session-token`: if present, this value should be used for as the session token
      - `s3.remote-signing-enabled`: if `true` remote signing should be performed as described in the `s3-signer-open-api.yaml` specification
```
Comment (stirupati) [inline open-api/rest-catalog-open-api.py:1101]: can we also add a key for expiration time for aws sts token ?

Comment (Fokko) [reply open-api/rest-catalog-open-api.py:1101]: @stirupati I would suggest splitting that out in a separate PR, so we can purely focus on GCS/ADLS for this PR.

File: open-api/rest-catalog-open-api.py [hunk 3]
```diff
@@ -1100,6 +1100,31 @@ class LoadTableResult(BaseModel):
...
+     - `gcs.encryption-key`: Option for customer-supplied AES256 key for server-side encryption of the blob
+     - `gcs.user-project`: Option for blob's billing user project. This option is used only if the blob's bucket has requester_pays flag enabled
+     - `gcs.channel.read.chunk-size-bytes`: Optional INT that sets the minimum size that will be read by a single RPC. Read data will be locally buffered until consumed
+     - `gcs.channel.write.chunk-size-byte`: Optional INT that sets the  minimum size that will be written by a single RPC. Written data will be buffered and only flushed upon reaching this size or closing the channel.
+     - `gcs.oauth2.token`: String representation of the access token used for temporary access.
+     - `gcs.oauth2.token-expires-at`: A LONG that represents the date in Epoch milliseconds of when the token expires.
+     - `gcs.no-auth`: Boolean to explicitly configure "no authentication" for testing purposes using a GCS emulator
+     - `gcs.delete.batch-size`: Optional INT that configures the batch size used when deleting multiple files from a given GCS bucket. Defaults to 50.
```
Comment (Fokko) [inline open-api/rest-catalog-open-api.py:1116]: Should we only limit this to configuration related to authentication? This is the case of AWS as well.

Comment (Buktoria) [reply open-api/rest-catalog-open-api.py:1116]: We could, but I would argue that we would still want those other configs documented in the spec somewhere for constancy sake for those working on their own FileIO implementations or Iceberg REST catalog implementations. 

If we limit what is documented in this section to auth, where do you propose we put other config that could be returned?

Comment (Buktoria) [reply open-api/rest-catalog-open-api.py:1116]: Any thoughts on the above @Fokko?

Comment (Fokko) [reply open-api/rest-catalog-open-api.py:1116]: Hey @Buktoria Thanks for pinging me. 

I think the more appropriate place is to have it under the `GcsFileIO`, [similar to the `S3FileIO`](https://iceberg.apache.org/docs/nightly/aws/#s3-fileio).

File: open-api/rest-catalog-open-api.py (hunk 2 above)
Comment (munendrasn) [reply open-api/rest-catalog-open-api.py:1101]: @Fokko created separate PR for the same https://github.com/apache/iceberg/pull/10873

**Pull request closed without merging**